File: front-veterinaria/src/app/Services/especie.service.ts
import { Injectable } from '@angular/core';
import { Especie, EspecieCustom, EspecieMapper } from '../Models/especie';
import { Observable } from 'rxjs';
import { API_URL } from '../Environments/routes';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class EspecieService {

  constructor(private http: HttpClient) { }

  findAll(): Observable<Especie[]> {
    return this.http.get<Especie[]>(API_URL.especies.find_all);
  }

  findAllCustom(): Observable<EspecieCustom[]> {
    return this.http.get<EspecieCustom[]>(API_URL.especies.find_all_custom);
  }

  findAllMapper(): Observable<EspecieMapper[]> {
    return this.http.get<EspecieMapper[]>(API_URL.especies.find_all_mapper);
  }

  findById(id: number): Observable<Especie> {
    return this.http.get<Especie>(API_URL.especies.find_by_id(id));
  }

  insert(especie: Especie): Observable<any> {
    return this.http.post<any>(API_URL.especies.insert, especie);
  }

  update(id: number, especie: Especie): Observable<any> {
    return this.http.put<any>(API_URL.especies.update(id), especie);
  }

  delete(id: number): Observable<any> {
    return this.http.delete<any>(API_URL.especies.delete(id));
  }

}

File: back-veterinaria/src/main/java/idat/proyecto/veterinaria/service/EspecieService.java
package idat.proyecto.veterinaria.service;

import org.springframework.http.ResponseEntity;

import idat.proyecto.veterinaria.entity.Especie;

public interface EspecieService {
	
	public abstract ResponseEntity<?> insert(Especie especie);
	public abstract ResponseEntity<?> update(Integer id, Especie especie);
	public abstract ResponseEntity<?> delete(Integer id);
	public abstract ResponseEntity<?> findById(Integer id);
	public abstract ResponseEntity<?> findAll();
	public abstract ResponseEntity<?> findAllCustom();
	public abstract ResponseEntity<?> findAllMapper();
}
